### PR TITLE
kodi: cleanup

### PR DESCRIPTION
**Pull request by aanderse**
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Some cleanup. Note that the first commit ~~does~~ did **NOT** cause a rebuild... until I changed the `name` attribute as suggested.

Add missing executable in `PATH`:
```
> /nix/store/pm1damnzkdr6551hyj7ysji8x42kv7ji-kodi-19.0/bin/kodi                                                                                                                                                                                                         
libva info: VA-API version 1.10.0
libva info: Trying to open /run/opengl-driver/lib/dri/nvidia_drv_video.so
libva info: Trying to open /usr/lib/dri/nvidia_drv_video.so
libva info: Trying to open /usr/lib32/dri/nvidia_drv_video.so
libva info: va_openDriver() returns -1
sh: nmblookup: command not found
sh: nmblookup: command not found
sh: nmblookup: command not found
sh: nmblookup: command not found
```

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by bhipple — APPROVED** [review]

**Comment by peterhoeg** [comment]
Any chance we can get libdvdcss libdvdnav libdvdread working properly instead? Ideally we build them separately (that was the idea behind `kodiDependency`) and then patch the cmake files to find them.

**Comment by aanderse** [comment]
Oh... what is the current issue? `kodi` seems to internally build them fine.

**Comment by peterhoeg** [comment]
Originally, we built everything from kodi in one go (kodi, their custom ffmpeg, assorted dvd libraries). That meant that everything took much longer than needed, so I split out ffmpeg first, so at least we didn't have to build that every time there was a change to kodi itself.

The plan (hope) was to do the same to the various other dependencies (the dvd libs), but I didn't get that working due to how the cmake files were done.

I don't know how much we gain from building those libraries separately in all fairness. They might be pretty quick, in which case this is perfectly fine.

**Comment by aanderse** [comment]
Thanks @peterhoeg I'll take a look.

File: pkgs/applications/video/kodi/default.nix [hunk 1]
```diff
@@ -57,41 +57,14 @@ let
...
-      };
-    } // attrs');
-
-  ffmpeg = kodiDependency rec {
-    name    = "FFmpeg";
-    version = "4.3.1";
-    rev     = "${version}-${rel}-Beta1";
-    sha256  = "1c5rwlxn6xj501iw7masdv2p6wb9rkmd299lmlkx97sw1kvxvg2w";
+  ffmpeg = stdenv.mkDerivation rec {
+    name = "kodi-ffmpeg-4.3.1";
```
Comment (SuperSandro2000) [inline pkgs/applications/video/kodi/default.nix:61]:
```suggestion
    pname = "kodi-ffmpeg";
    version = "4.3.1";
```

**Comment by aanderse** [comment]
> I don't know how much we gain from building those libraries separately in all fairness. They might be pretty quick, in which case this is perfectly fine.

I just built 2 out of the 3 libraries and it was a ~5 seconds for each. If the reason was speed I think we're fine to simplify things and have the `kodi` build system take care of it for us :+1:

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115158` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
15 packages built:
</summary>
<ul>
<li>kodi</li>
<li>kodi-gbm</li>
<li>kodi-wayland</li>
<li>kodiGBM</li>
<li>kodiPlain (kodiPlugins.kodi)</li>
<li>kodiPlainWayland</li>
<li>kodiPlugins.inputstream-adaptive</li>
<li>kodiPlugins.joystick</li>
<li>kodiPlugins.kodi-platform</li>
<li>kodiPlugins.pvr-hdhomerun</li>
<li>kodiPlugins.pvr-hts</li>
<li>kodiPlugins.pvr-iptvsimple</li>
<li>kodiPlugins.steam-controller</li>
<li>kodiPlugins.vfs-libarchive</li>
<li>kodiPlugins.vfs-sftp</li>
</ul>
</details>

**Review by minijackson — APPROVED** [review]
Both kodi and kodi-wayland works great, tested with the sftp addon

---

Result of `nix-review pr 115158` [1](https://github.com/Mic92/nix-review)
<details>
  <summary>15 package were built:</summary>

  - kodi (kodiPlugins.kodi)
  - kodi-gbm (kodiPlugins.kodi)
  - kodi-wayland (kodiPlugins.kodi)
  - kodiGBM (kodiPlugins.kodi)
  - kodiPlain (kodiPlugins.kodi)
  - kodiPlainWayland (kodiPlugins.kodi)
  - kodiPlugins.inputstream-adaptive (kodiPlugins.kodi)
  - kodiPlugins.joystick (kodiPlugins.kodi)
  - kodiPlugins.kodi-platform (kodiPlugins.kodi)
  - kodiPlugins.pvr-hdhomerun (kodiPlugins.kodi)
  - kodiPlugins.pvr-hts (kodiPlugins.kodi)
  - kodiPlugins.pvr-iptvsimple (kodiPlugins.kodi)
  - kodiPlugins.steam-controller (kodiPlugins.kodi)
  - kodiPlugins.vfs-libarchive (kodiPlugins.kodi)
  - kodiPlugins.vfs-sftp (kodiPlugins.kodi)
</details>

---

Result of `nixpkgs-hammer kodiPlain` [2](https://github.com/jtojnar/nixpkgs-hammering)

<details>
  <summary>Output:</summary>

```
When evaluating attribute ‘kodiPlain’:
warning: build-tools-in-build-inputs
unzip is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near /home/minijackson/.cache/nix-review/pr-115158-1/nixpkgs/pkgs/applications/video/kodi/default.nix:119:5:
    |
119 |     buildInputs = [
    |     ^
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: build-tools-in-build-inputs
zip is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near /home/minijackson/.cache/nix-review/pr-115158-1/nixpkgs/pkgs/applications/video/kodi/default.nix:119:5:
    |
119 |     buildInputs = [
    |     ^
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: attribute-ordering
The lists of dependencies, including the attribute “nativeBuildInputs” should preferably come before lists of dependencies’ “buildInputs” attribute in the expression.

Near /home/minijackson/.cache/nix-review/pr-115158-1/nixpkgs/pkgs/applications/video/kodi/default.nix:119:5:
    |
119 |     buildInputs = [
    |     ^
Near /home/minijackson/.cache/nix-review/pr-115158-1/nixpkgs/pkgs/applications/video/kodi/default.nix:166:5:
    |
166 |     nativeBuildInputs = [
    |     ^
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-ordering.md
warning: attribute-ordering
The attributes for overriding default phases, including the attribute “postPatch” should preferably come before attributes for overriding default phases’ “preConfigure” attribute in the expression.

Near /home/minijackson/.cache/nix-review/pr-115158-1/nixpkgs/pkgs/applications/video/kodi/default.nix:204:5:
    |
204 |     preConfigure = ''
    |     ^
Near /home/minijackson/.cache/nix-review/pr-115158-1/nixpkgs/pkgs/applications/video/kodi/default.nix:216:5:
    |
216 |     postPatch = ''
    |     ^
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-ordering.md
warning: attribute-ordering
The attributes for enabling off-by-default phases, including the attribute “doInstallCheck” should preferably come before attributes for overriding default phases’ “postInstall” attribute in the expression.

Near /home/minijackson/.cache/nix-review/pr-115158-1/nixpkgs/pkgs/applications/video/kodi/default.nix:221:5:
    |
221 |     postInstall = ''
    |     ^
Near /home/minijackson/.cache/nix-review/pr-115158-1/nixpkgs/pkgs/applications/video/kodi/default.nix:235:5:
    |
235 |     doInstallCheck = true;
    |     ^
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-ordering.md
warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near /home/minijackson/.cache/nix-review/pr-115158-1/nixpkgs/pkgs/applications/video/kodi/default.nix:246:7:
    |
246 |       license     = licenses.gpl2;
    |       ^
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
```

</details>

File: pkgs/applications/video/kodi/default.nix (hunk 1 above)
Comment (aanderse) [reply pkgs/applications/video/kodi/default.nix]: Done.

**Pull request merged**